packages/ubuntu_desktop_installer/lib/services/desktop_service.dart

```diff
@@ -15,29 +28,106 @@ abstract class DesktopService {
 class GnomeService implements DesktopService {
   /// Creates a GNOME settings instance using the given GSettings as a backend
   /// for storing the settings.
-  GnomeService([@visibleForTesting GSettings? gsettings])
-      : _gsettings = gsettings ?? GSettings('org.gnome.desktop.interface');
+  GnomeService({
+    @visibleForTesting GSettings? dingSettings,
+    @visibleForTesting GSettings? interfaceSettings,
+    @visibleForTesting GSettings? mediaHandlingSettings,
+    @visibleForTesting GSettings? sessionSettings,
+    @visibleForTesting GSettings? screensaverSettings,
+  })  : _dingSettings =
+            dingSettings ?? GSettings('org.gnome.shell.extensions.ding'),
+        _interfaceSettings =
+            interfaceSettings ?? GSettings('org.gnome.desktop.interface'),
+        _mediaHandlingSettings = mediaHandlingSettings ??
+            GSettings('org.gnome.desktop.media-handling'),
+        _screensaverSettings =
+            screensaverSettings ?? GSettings('org.gnome.desktop.screensaver'),
+        _sessionSettings =
+            sessionSettings ?? GSettings('org.gnome.desktop.session');
 
-  final GSettings _gsettings;
+  final GSettings _dingSettings;
+  final GSettings _interfaceSettings;
+  final GSettings _mediaHandlingSettings;
+  final GSettings _screensaverSettings;
+  final GSettings _sessionSettings;
+
+  final restoreSettings = <Future<void> Function()>[];
+
+  @override
+  Future<void> disableAutoMounting() async {
+    logger.debug('Disabling automounting');
+    final previousAutoMount = await _mediaHandlingSettings.get('automount');
+    final previousAutoMountOpen =
+        await _mediaHandlingSettings.get('automount-open');
+    final previousAutoRunNever =
+        await _mediaHandlingSettings.get('autorun-never');
+    final previousShowVolumes = await _dingSettings.get('show-volumes');
+    final previousShowNetworkVolumes =
+        await _dingSettings.get('show-network-volumes');
+    await _mediaHandlingSettings.set('automount', const DBusBoolean(false));
+    await _mediaHandlingSettings.set(
+        'automount-open', const DBusBoolean(false));
+    await _mediaHandlingSettings.set('autorun-never', const DBusBoolean(true));
+    await _dingSettings.set('show-volumes', const DBusBoolean(false));
+    await _dingSettings.set('show-network-volumes', const DBusBoolean(false));
+    restoreSettings.add(() async {
+      await _mediaHandlingSettings.set('automount', previousAutoMount);
```

Comment:
would reset() work or are these explicitly set in the beginning?

Reply:
I just mimicked what ubiquity is doing, but I guess it's a little overkill. Does anyone fiddle with screensaver and automounting settings in a live system, run the installer, and then decide to continue using the live system, expecting all their precious settings are restored?